src/HelperFunctions/Helpers.cs

```diff
@@ -548,23 +548,27 @@ public static bool IsServiceRunning(string name)
             int size = 0;
 
             bool success = SetupApi.SetupCopyOEMInf(infPath, "", SetupApi.SPOST.NONE, SetupApi.SP_COPY.NOOVERWRITE, IntPtr.Zero, 0, ref size, IntPtr.Zero);
-
+            bool bcontinue = true;
+            reboot = false;
             if (!success)
             {
                 
                 int error = Marshal.GetLastWin32Error();
                 Trace.WriteLine("Unable to update driver - code " + error.ToString());
                 if ((error == 0) || (error == 0x50))
                 {
-                    Trace.WriteLine("Driver already installed");
-                    if (Helpers.IsServiceRunning(name))
-                    {
-                        reboot = false;
-                        return;
-                    }
+                    // We still try to install the driver even the OEMInf file has already been copied,

```

Comment:
I'm concerned that this change might bring back the issue fixed by CA-248127 

The idea is that we don't want to reinstall a working driver which is already in place (because if an update contains no new drivers, but just a new guest agent, we don't want to force a reboot)

(You can test this by building an installer, installing it, then building an installer with a higher build number (but everything else the same) and installing it.  The desired behaviour is that we need no reboots before the install succeeds.  Could you confirm that this behaviour is retained by your changes?)

Reply:
good point, I will try to verify it.

Reply:
hi ben,
    I verified my builds and the behavior is retained, which means reboot is not required. 
   The test build is 
https://repo.citrite.net/xs-local-build/win-installer/lindev/win-installer-1124(5/6/7).